src/System.Reflection/tests/PropertyInfo/PropertyInfo_GetSetValueTests.cs

```diff
@@ -0,0 +1,374 @@
+
+            Assert.Equal("992f1someotherstring", obj.setValue);
+
+            pi.SetValue(obj, "pw", new Object[] { 99, 2, h, "SOME  string" });
+
+            Assert.Equal("992SOME  string1pw", obj.setValue);
+        }
+
+        //
+        // Negative Tests for PropertyInfo
+        //
+
+        [Theory]
+        [InlineData("PropertyC", typeof(MyCoVariantTest), new Object[] { 1, "2", 3 })]
+        [InlineData("PropertyC", typeof(MyCoVariantTest), null)]
+        public static void TestGet_ThrowsTargetParameterCountException(string propertyName, Type type, Object[]testObj)
+        {
+            Object obj = Activator.CreateInstance(type);
+            PropertyInfo pi = GetProperty(type, propertyName);
+
+            Assert.Equal(pi.Name, propertyName);
+
+            Assert.Throws<TargetParameterCountException>(() =>
+            {
+                var value = pi.GetValue(obj, testObj);
+            });
+        }
+
+        [Theory]
+        [InlineData("Property1", typeof(LaterClass), null)]
+        [InlineData("PropertyC", typeof(MyCoVariantTest), new Object[] { "1", "2" })]
+        public static void TestGet_ThrowsArgumentException(string propertyName, Type type, Object[] testObj)
+        {
+            Object obj = Activator.CreateInstance(type);
+            PropertyInfo pi = GetProperty(type, propertyName);
+
+            Assert.Equal(pi.Name, propertyName);
+
+            Assert.Throws<ArgumentException>(() =>
+            {
+                var value = pi.GetValue(obj, testObj);
+            });
+        }
+
+
+        //Verify PropertyInfo.GetValue throws Exception
+        [Fact]
+        public static void TestGet_ThrowsException()
+        {
+            string propertyName = "PropertyC";
+            Object obj = Activator.CreateInstance(typeof(MyCoVariantTest));
+            PropertyInfo pi = GetProperty(typeof(MyCoVariantTest), propertyName);
+
+            Assert.Equal(pi.Name, propertyName);
+
+            // In Win8p instead of TargetException , generic Exception is thrown
+            // Refer http://msdn.microsoft.com/en-us/library/b05d59ty.aspx
+            Assert.ThrowsAny<Exception>(() => pi.GetValue(null, new Object[] { "1", "2" }));
+        }
+
+
+
+        [Theory]
+        [InlineData("NoSetterProp", 100, null)]
+        [InlineData("Item", "pw", new Object[] { 99, 2, "hello", "SOME  string" })]
+        [InlineData("Item", 100, new Object[] { 99, 2, new string[] { "hello" }, "SOME  string" })]
+        public static void TestSet_ThrowsArgumentException(string propertyName, Object setValue, Object[] index)
+        {
+            TypeWithProperties obj = new TypeWithProperties();
+            PropertyInfo pi = GetProperty(typeof(TypeWithProperties), propertyName);
+
+            Assert.Equal(pi.Name, propertyName);
+
+            Assert.Throws<ArgumentException>(() =>
+            {
+                pi.SetValue(obj, setValue, index);
+            });
```

Comment:
Nit: C# lambdas can include statement expressions, so as earlier, this can also be condensed to:
```C#
Assert.Throws<ArgumentException>(() => pi.SetValue(obj, setValue, index));
```